src/charm.py

```diff
@@ -318,10 +318,13 @@ def _configure_legacy(self, charm_state: CharmState) -> None:
     def _configure_haproxy_route(self, charm_state: CharmState) -> None:
         """Configure the haproxy route relation."""
         haproxy_route_requirers_information = HaproxyRouteRequirersInformation.from_provider(
-            self.haproxy_route_provider,
-            self.haproxy_route_tcp_provider,
-            typing.cast(typing.Optional[str], self.model.config.get("external-hostname")),
-            self._get_peer_units_address(),
+            haproxy_route=self.haproxy_route_provider,
+            haproxy_route_tcp=self.haproxy_route_tcp_provider,
+            external_hostname=typing.cast(
+                typing.Optional[str], self.model.config.get("external-hostname")
+            ),
+            peers=self._get_peer_units_address(),
+            ca_certs_configured=HAPROXY_CAS_FILE.exists(),
```

Comment:
should we also include `bool(self.recv_ca_cert.get_all_certificates())` to check if a CA cert was transferred?